Dart/src/com/jetbrains/lang/dart/resolve/DartResolver.java

```diff
@@ -189,6 +185,33 @@ else if (!lValue && component.isGetter()) {
     });
   }
 
+  /**
+   * Find the region with the given offset in the given list of sorted regions.
+   * Returns the found region or null.
+   */
+  @Nullable
+  private static PluginNavigationRegion findRegionAtOffset(List<PluginNavigationRegion> regions, int offset) {
```

Comment:
O(ln) is better that O(n) :)
Did you see not zero time for this search in profiler? If needed we can make it O(1) by introducing an array of regions with regions.getLast().getOffset() size that would map offset to region.